ring_theory/algebra.lean

```diff
@@ -0,0 +1,495 @@
+/-
+Copyright (c) 2018 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+
+Algebra over Commutative Ring (under category)
+-/
+
+import data.polynomial
+import linear_algebra.multivariate_polynomial
+import linear_algebra.tensor_product
+import ring_theory.subring
+
+universes u v w u₁ v₁
+
+open lattice
+
+local infix ` ⊗ `:100 := tensor_product
+
+structure algebra.core (R : Type u) (A : Type v) [comm_ring R] [comm_ring A] :=
```

Comment:
This should have a comment explaining its purpose and relationship to `algebra`.

Reply:
Like how `module.core` has a comment above it explaining its purpose and relationship to `module`?

Reply:
I think that those Coq files Assia showed us this morning at Lean Together, which all seem to start with 50 lines of comment, put mathlib to shame.

Reply:
There is no reason not to start filling mathlib with instructional comments. Would PR's of this nature be of interest?

Reply:
@kbuzzard Certainly. I did a docs sprint a while ago and it seemed to be well received - we just need people to actually write it. You are probably in a good position to do it, if you have the time.